SystemServices/SystemServices.cpp

```diff
@@ -4079,14 +4080,14 @@ namespace WPEFramework {
                                       return strcasecmp(C_STR(t.first), C_STR(str)) == 0;
                                   });
                 if (it != FwFailReasonFromText.end())
-                    failReason = it->second;
+                    flReason = it->first;
                 else if (!str.empty())
                     LOGWARN("Unrecognised FailureReason!");
             } else {
                 LOGINFO("Could not read file %s", FWDNLDSTATUS_FILE_NAME);
             }
 
-            response["failReason"] = FwFailReasonToText.at(failReason);
+            response["failReason"] = flReason;
```

Comment:
please revert. you should not make changes in SystemServices. this api may deliberately ignore "Invalid Request" because "Invalid Request" doesn't relate to firmware failures

Reply:
reverted